src/TypeFormatter/LiteralUnionTypeFormatter.ts

```diff
@@ -72,13 +73,23 @@ export class LiteralUnionTypeFormatter implements SubTypeFormatter {
 export function isLiteralUnion(type: UnionType): boolean {
     return type
         .getFlattenedTypes()
-        .every((item) => item instanceof LiteralType || item instanceof NullType || item instanceof StringType);
+        .every(
+            (item) =>
+                item instanceof LiteralType ||
+                item instanceof NullType ||
+                item instanceof StringType ||
+                item instanceof EnumType,
+        );
 }
 
-function getLiteralValue(value: LiteralType | NullType): LiteralValue | null {
-    return value instanceof LiteralType ? value.getValue() : null;
+function getLiteralValues(value: LiteralType | EnumType | NullType): readonly (LiteralValue | null)[] {
+    return value instanceof LiteralType ? [value.getValue()] : value instanceof EnumType ? value.getValues() : [null];
```

Comment:
if else/please

Reply:
please also add a test with a literal/enum + something non literal to ensure its still `anyOf`?